matrix-sdk/src/main/java/org/matrix/androidsdk/rest/api/RoomsApi.java

```diff
@@ -146,23 +147,17 @@
     @POST("rooms/{roomId}/invite")
     Call<Void> invite(@Path("roomId") String roomId, @Body Map<String, String> params);
 
-    /**
-     * Join the given room.
-     *
-     * @param roomId  the room id
-     * @param content the request body
-     */
-    @POST("rooms/{roomId}/join")
-    Call<Void> join(@Path("roomId") String roomId, @Body JsonObject content);
-
     /**
      * Join the room with a room id or an alias.
      *
      * @param roomAliasOrId a room alias (or room id)
+     * @param viaServers    The servers to attempt to join the room through. One of the servers must be participating in the room.
      * @param params        the extra join param
      */
     @POST("join/{roomAliasOrId}")
-    Call<RoomResponse> joinRoomByAliasOrId(@Path("roomAliasOrId") String roomAliasOrId, @Body Map<String, Object> params);
+    Call<RoomResponse> joinRoomByAliasOrId(@Path("roomAliasOrId") String roomAliasOrId,
+                                           @Query("server_name") List<String> viaServers,
```

Comment:
Does this annotation create queries like `server_name=server1.com&server_name=server2.com` when they are several `viaServers`?

Reply:
Yes, see https://square.github.io/retrofit/2.x/retrofit/index.html?retrofit2/http/Query.html
"Passing a List or array will result in a query parameter for each non-null item."